client/src/Components/local/atoms/mapping-panel/helpers/icon-renderer.tsx

```diff
@@ -12,23 +16,24 @@ type IconRendererProps = {
 }
 
 const IconRenderer: React.FC<IconRendererProps> = ({ feature, checked, onClick }) => {
-  const iconRef = useRef<HTMLDivElement>(null)
- 
-  useEffect(() => {
-    // TODO: reserch how to render shape in this line
+  const iconElm = useMemo(() => {
+    if (!feature) {
+      return
+    }
     const icon = new ms.Symbol(feature.properties?.sidc, { size: 20 })
     const health = feature.properties?.health
 
     const healthColor = calculateHealthColor(health)
 
-    if (iconRef.current) {
```

Comment:
Hey @lebaphi - I tried again and again to make this fix myself, I just couldn't manage the tsx constructs for it. 

Thanks!